Source/ResponseSerialization.swift

```diff
@@ -144,25 +143,57 @@ extension DataRequest {
         completionHandler: @escaping (DataResponse<Serializer.SerializedObject>) -> Void)
         -> Self
     {
-        internalQueue.addOperation {
-            self.serializationQueue.async {
-                let start = CFAbsoluteTimeGetCurrent()
-                let result = Result { try responseSerializer.serialize(request: self.request,
-                                                                       response: self.response,
-                                                                       data: self.data,
-                                                                       error: self.error) }
-                let end = CFAbsoluteTimeGetCurrent()
-
-                let response = DataResponse(request: self.request,
-                                            response: self.response,
-                                            data: self.data,
-                                            metrics: self.metrics,
-                                            serializationDuration: (end - start),
-                                            result: result)
-
-                self.eventMonitor?.request(self, didParseResponse: response)
-
-                (queue ?? .main).async { completionHandler(response) }
+        appendResponseSerializer {
+            let start = CFAbsoluteTimeGetCurrent()
+            let result = Result { try responseSerializer.serialize(request: self.request,
+                                                                   response: self.response,
+                                                                   data: self.data,
+                                                                   error: self.error) }
+            let end = CFAbsoluteTimeGetCurrent()
+
+            let queue: DispatchQueue = queue ?? .main
+            let response = DataResponse(request: self.request,
+                                        response: self.response,
+                                        data: self.data,
+                                        metrics: self.metrics,
+                                        serializationDuration: (end - start),
+                                        result: result)
+
+            self.eventMonitor?.request(self, didParseResponse: response)
+
+            if let serializerError = result.error, let delegate = self.delegate {
```

Comment:
It seems like this would be better as a `guard`, since the `else` would have a simple call to `self.responseSerializerDidComplete { queue.async { completionHandler(response) } }`.

Reply:
Will update.

Reply:
Switched to `guard` in fea5fc86.